Simulator/Edges/AllEdges.h

```diff
@@ -198,26 +198,26 @@ class AllEdges {
 #endif   // defined(USE_GPU)
 
    ///  The location of the edge.
-   int *sourceVertexIndex_;
+   vector<int> sourceVertexIndex_;
 
    ///  The coordinates of the summation point.
-   int *destVertexIndex_;
+   vector<int> destVertexIndex_;
 
    ///   The weight (scaling factor, strength, maximal amplitude) of the edge.
-   BGFLOAT *W_;
+   vector<BGFLOAT> W_;
 
    ///  This edge's summation point's address.
-   BGFLOAT **summationPoint_;
+   vector<BGFLOAT *> summationPoint_;
 
    ///   Synapse type
-   edgeType *type_;
+   vector<edgeType> type_;
 
    ///  The boolean value indicating the entry in the array is in use.
-   bool *inUse_;
+   unique_ptr<bool[]> inUse_;
 
    ///  The number of (incoming) edges for each vertex.
    ///  Note: Likely under a different name in GpuSim_struct, see edge_count. -Aaron
-   BGSIZE *edgeCounts_;
+   vector<BGSIZE> edgeCounts_;
```

Comment:
I'm a little concerned about initialization. I _think_ that `resize()` will just initialize the newly added values in a vector, so that would mean that the initially created entries in the default sized vectors might not be initialized correctly.

Reply:
Used assign to assign default values.